packages/flutter/lib/src/widgets/binding.dart

```diff
@@ -1229,6 +1229,12 @@ mixin WidgetsBinding on BindingBase, ServicesBinding, SchedulerBinding, GestureB
   /// The `rootWidget` widget provided to this method must not already be
   /// wrapped in a [View].
   Widget wrapWithDefaultView(Widget rootWidget) {
+    assert(
+      platformDispatcher.implicitView != null,
+      'The implicitView is null. Cannot create a default View from `runApp`. '
```

Comment:
This error uses jargon that most Flutter developers won't know. This might frustrate users.

Perhaps we should add more context? Maybe something like:

> `runApp` requires an implicit view.
>
> `runApp` renders the widget tree into a view provided by the platform,
> called the "implicit view". However, the platform did not provide an
> implicit view.
>
> Try using `runWidget` instead of `runApp`.
>
> See: https://flutter.dev/to/web-runwidget

Reply:
Since it's a public API, I think we should not assume it 100% comes from `runApp`. Maybe

> The app requested an implicit view, but none is found.
>
> This is likely because the app called `runApp` to start up, which expects the platform to provide a default view to render into ("the implicit view"), while the engine is configured to let the app handle view creation (which allows creating multiple views.) 
>
> Try using `runWidget` instead of `runApp`. See also https://flutter.dev/to/web-runwidget

Reply:
@loic-sharma @dkwingsmt I ended up with something like this:

> The app is attempting to render into a default View that isn't available (`platformDispatcher.implicitView == null`).
> This is likely because the platform has multi-view mode enabled, but the app is calling `runApp` to render its root Widget.
> Try using `runWidget` instead of `runApp` to start your app. `runWidget` allows you to provide a `View` widget, without requiring a default on the platform.
> See: https://flutter.dev/to/web-multiview-runwidget

(Tried to use lingo closer to that in the [wrapWithDefaultView method docs](https://api.flutter.dev/flutter/widgets/WidgetsBinding/wrapWithDefaultView.html).)

Reply:
I slightly prefer Tong's wording.

Some context:

1. On desktop, the north star is that the app will call `runApp` even though there's no implicit view. `runApp` will know how to create a window to host the widget tree.
2. On desktop, you can have multiple views while also having an implicit view. Thus, on desktop you can call `runApp` even if you have multi-view mode enabled

What do y'all think about this? I tried to merge bits from all our messages :)

> The app requested an implicit view, but the platform doesn't have one.
> 
> This is likely because the app called `runApp` to render its root widget,
> which expects the platform to provide a default view to render into ("the implicit view").
> However, the platform likely has multi-view mode enabled, which does not create a default "implicit" view.
> 
> Try using `runWidget` instead of `runApp` to start your app. `runWidget` allows you to provide a `View` widget, without requiring a default view.
> See: https://flutter.dev/to/web-multiview-runwidget

Reply:
Looks good!

I think checking it in wrapWithDefaultView is better than runApp since wrapWithDefaultView is also a public API and users can still call it incorrectly (which is a user error and you shouldn't assert on).

Reply:
I'm happy with this too, I applied @loic-sharma's suggestion! If @dkwingsmt is happy, I'm happy too, let's `autosubmit` this  :P

Reply:
> `runApp` will know how to create a window to host the widget tree.
> ...
> you can have multiple views while also having an implicit view

We need to discuss this elsewhere, I'm interested :P